include/alpaka/core/ConcurrentExecPool.hpp

```diff
@@ -73,7 +73,7 @@ namespace alpaka
                 {
                     std::lock_guard<std::mutex> lk(m_Mutex);
 
-                    std::queue<T>::push(std::forward<T>(t));
+                    std::queue<T>::push(t);
```

Comment:
No, because `push` is no template, `T && t` is no 'universal reference' (representing either `&` or `&&`) but is forced to be a rvalue reference (`&&`). This means that `t` can only ever be moved into the `push` function. This means that the caller will always give up his reference and will transfer the ownership to the `push` function which will then transfer it to the internal queue via `forward`. At the call side, this can be reognized by the `std::move(upTaskPackage)`. It will not compile without the `std::move`.